### PR TITLE
Introduce encryptedWalletsStore

### PR DESCRIPTION
**Pull request by Sajma27**
# Summary
A description of the changes proposed in the pull request.

Fixes #number

## Checklist
- [ ] Change is small and easy to review (please split big changes into multiple PRs)
- [ ] Change is consistent with architecture
- [ ] Change is consistent with test architecture
- [ ] Change is consistent with naming conventions
- [ ] New code is covered with tests
- [ ] Tests related to old code are updated
- [ ] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-relayer/test/integration/sql/EncryptedWalletsStore.test.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import {expect} from 'chai';
+import {EncryptedWallet, TEST_WALLET_JSON} from '@unilogin/commons';
+import {getKnexConfig} from '../../testhelpers/knex';
+import {EncryptedWalletsStore} from '../../../src/integration/sql/services/EncryptedWalletsStore';
+
+describe('UNIT: EncryptedWalletsStore', () => {
```
Comment (xitronix) [inline universal-login-relayer/test/integration/sql/EncryptedWalletsStore.test.ts:6]: INT

File: universal-login-commons/src/core/constants/test.ts [hunk 2]
```diff
@@ -142,3 +142,7 @@ export const TEST_GAS_PRICES = {
     timeEstimation: '30',
   },
 } as unknown as GasPriceSuggestion;
+
+export const TEST_ENCRYPTED_WALLET_JSON = '{"address":"76010666821749b24145d6d792a3038a6de0cd71","id":"dc134bc2-9d47-4e73-990f-c9abeee94c41","version":3,"Crypto":{"cipher":"aes-128-ctr","cipherparams":{"iv":"7c4e7269f68a4b93337fd6f17d956735"},"ciphertext":"6897e05bf1088a1e83b5c8ca3648b55cbc417661de660d2a30da65223b5f2fd0","kdf":"scrypt","kdfparams":{"salt":"a8e44cd1357b86dcf387c29dd09f4f3b3d86ac6d411f3f88a069ef5f830b3c2e","n":131072,"dklen":32,"p":1,"r":8},"mac":"cc555dbde8816b36fbb4368fe3e9045c5a516a9028c51ada41860258b4da8ede"},"x-ethers":{"client":"ethers.js","gethFilename":"UTC--2020-07-09T14-31-30.0Z--76010666821749b24145d6d792a3038a6de0cd71","mnemonicCounter":"0d2f10e922bc048cdef52a543abbad3c","mnemonicCiphertext":"5eb6e83c13d073136f3125bbe47ee553","path":"m/44\'/60\'/0\'/0/0","version":"0.1"}}';
```
Comment (JustynaBroniszewska) [inline universal-login-commons/src/core/constants/test.ts:146]: We should make it typical JSON, not string.

File: universal-login-relayer/src/integration/sql/services/EncryptedWalletsStore.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import {EncryptedWallet} from '@unilogin/commons';
+
+export class EncryptedWalletsStore {
+  private tableName = 'encrypted_wallets';
+
+  constructor(private database: Knex) {
+  }
+
+  async add(encryptedWallet: EncryptedWallet) {
+    return (await this.database(this.tableName)
```
Comment (JustynaBroniszewska) [inline universal-login-relayer/src/integration/sql/services/EncryptedWalletsStore.ts:11]: Here wallet JSON should be stringified

File: universal-login-relayer/test/integration/sql/EncryptedWalletsStore.test.ts (hunk 1 above)
Comment (xitronix) [reply universal-login-relayer/test/integration/sql/EncryptedWalletsStore.test.ts:6]: Add test with failure when adding duplicated email

Comment (Sajma27) [reply universal-login-relayer/test/integration/sql/EncryptedWalletsStore.test.ts]: Done

File: universal-login-commons/src/core/constants/test.ts (hunk 2 above)
Comment (Sajma27) [reply universal-login-commons/src/core/constants/test.ts]: Done

File: universal-login-relayer/src/integration/sql/services/EncryptedWalletsStore.ts (hunk 3 above)
Comment (Sajma27) [reply universal-login-relayer/src/integration/sql/services/EncryptedWalletsStore.ts:12]: Done

File: universal-login-commons/src/core/constants/test.ts [hunk 4]
```diff
@@ -142,3 +142,7 @@ export const TEST_GAS_PRICES = {
     timeEstimation: '30',
   },
 } as unknown as GasPriceSuggestion;
+
+export const TEST_ENCRYPTED_WALLET_JSON = JSON.parse('{"address":"76010666821749b24145d6d792a3038a6de0cd71","id":"dc134bc2-9d47-4e73-990f-c9abeee94c41","version":3,"Crypto":{"cipher":"aes-128-ctr","cipherparams":{"iv":"7c4e7269f68a4b93337fd6f17d956735"},"ciphertext":"6897e05bf1088a1e83b5c8ca3648b55cbc417661de660d2a30da65223b5f2fd0","kdf":"scrypt","kdfparams":{"salt":"a8e44cd1357b86dcf387c29dd09f4f3b3d86ac6d411f3f88a069ef5f830b3c2e","n":131072,"dklen":32,"p":1,"r":8},"mac":"cc555dbde8816b36fbb4368fe3e9045c5a516a9028c51ada41860258b4da8ede"},"x-ethers":{"client":"ethers.js","gethFilename":"UTC--2020-07-09T14-31-30.0Z--76010666821749b24145d6d792a3038a6de0cd71","mnemonicCounter":"0d2f10e922bc048cdef52a543abbad3c","mnemonicCiphertext":"5eb6e83c13d073136f3125bbe47ee553","path":"m/44\'/60\'/0\'/0/0","version":"0.1"}}');
```
Comment (xitronix) [inline universal-login-commons/src/core/constants/test.ts:146]: Can we do readable json?

Comment (JustynaBroniszewska) [reply universal-login-commons/src/core/constants/test.ts:146]: Just do something like: 
```ts
const TEST_ENCRYPTED_WALLET_JSON = {
    address: "929929292",
    id: "djdududuud",
    version: "3",
    ....
}
```

File: universal-login-commons/src/core/models/EncryptedWallet.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export interface EncryptedWallet {
+  email: string;
+  ensName: string;
+  walletJSON: JSON;
```
Comment (JustynaBroniszewska) [inline universal-login-commons/src/core/models/EncryptedWallet.ts:4]: Add new type `WalletJSON` 
```ts
interface WalletJSON {
address: string,
version: string
...
}
```

File: universal-login-commons/src/core/models/EncryptedWallet.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+export interface CipherParams {
+  iv: string;
+}
+
+export interface Crypto {
+  cipher: string;
+  cipherparams: CipherParams;
+  ciphertext: string;
+  kdf: string;
+  kdfparams: Record<string, any>;
```
Comment (JustynaBroniszewska) [inline universal-login-commons/src/core/models/EncryptedWallet.ts:18]: `kdfparams: KdfParams`

Comment (Sajma27) [reply universal-login-commons/src/core/models/EncryptedWallet.ts]: Done

**Pull request merged**